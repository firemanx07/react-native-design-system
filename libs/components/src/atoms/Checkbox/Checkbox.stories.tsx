import { ComponentStory, ComponentMeta } from '@storybook/react';
import React from 'react';

import { StorybookScreen, StorybookSection } from '../../storybook';
import {
  Checkbox as CheckboxBase,
  PropsType as CheckboxProps,
} from './Checkbox';

const Story: ComponentMeta<typeof CheckboxBase> = {
  component: CheckboxBase,
  title: 'Atoms/Checkbox',
};
export default Story;

const Template: ComponentStory<typeof CheckboxBase> = (args: CheckboxProps) => (
  <StorybookScreen>
    <StorybookSection title="CheckBox:">
      <CheckboxBase {...args} />
    </StorybookSection>

    <StorybookSection title="Disabled:">
      <CheckboxBase {...args} disabled />
    </StorybookSection>
    <StorybookSection title="inactive:">
      <CheckboxBase {...args} disabled={false} isChecked={false} />
    </StorybookSection>
  </StorybookScreen>
);

const parameters = {
  controls: {
    include: ['size', 'isChecked', 'disabled'],
  },
};

const defaultArgs: CheckboxProps = {
  size: 20,
  isChecked: true,
  disabled: false,
  onPress: () => {},
};

export const Checkbox = Template.bind({});
Checkbox.parameters = parameters;
Checkbox.args = {
  ...defaultArgs,
};
