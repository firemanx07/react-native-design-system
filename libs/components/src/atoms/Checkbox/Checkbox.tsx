import { styled } from '@proxym/themes';
import React, { memo } from 'react';
import { StyleProp, ViewStyle } from 'react-native';

import { TestIDType } from '../../types';
import { CheckDisabledIcon, CheckIcon } from '../Icons';

export type PropsType = {
  isChecked?: boolean;
  onPress?: () => void;
  size?: number;
  style?: StyleProp<ViewStyle>;
  disabled?: boolean;
} & TestIDType;

const BORDER_RADUIS_RATIO = 0.31;
export const Checkbox = ({
  isChecked,
  onPress,
  size = 20,
  style,
  testID,
  disabled,
}: PropsType) => {
  return (
    <Container
      size={size}
      isChecked={isChecked}
      style={style}
      onPress={onPress}
      disabled={!!disabled || !onPress}
      testID={testID}
    >
      {!disabled ? (
        (!!isChecked && <CheckboxIcon />) || <EmptyCheckboxIcon size={size} />
      ) : (
        <CheckboxDisabledIcon />
      )}
    </Container>
  );
};

type StyledCommonType = {
  size: number;
  isChecked?: boolean;
};

const Container = styled.TouchableOpacity<StyledCommonType>`
  height: ${({ size }) => size}px;
  width: ${({ size }) => size}px;
  justify-content: center;
  align-items: center;
`;
const EmptyCheckboxIcon = styled.View<StyledCommonType>`
  border-width: 2px;
  height: 100%;
  width: 100%;
  border-radius: ${({ size }) => size * BORDER_RADUIS_RATIO}px;
  border-color: ${({ theme }) => theme.ds.colors.grayDark};
`;

const CheckboxIcon = styled(CheckIcon).attrs<StyledCommonType>(({ theme }) => ({
  fill: theme.ds.colors.secondary,
  height: '100%',
  width: '100%',
}))``;
const CheckboxDisabledIcon = styled(CheckDisabledIcon).attrs(({ theme }) => ({
  fill: theme.ds.colors.grayDark,
  height: '100%',
  width: '100%',
}))``;

export default memo(Checkbox);
