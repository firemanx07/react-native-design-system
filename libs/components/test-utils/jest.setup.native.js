import { Animated } from 'react-native';

require('react-native-reanimated/lib/reanimated2/jestUtils').setUpTests();
require('react-native-gesture-handler/jestSetup.js');

console.warn = () => {
  //this is mock
};

Animated['timing'] = (value, config) => ({
  start: callback => {
    value.setValue(config.toValue);
    callback && callback({ finished: true });
  },
  stop: jest.fn(),
});
