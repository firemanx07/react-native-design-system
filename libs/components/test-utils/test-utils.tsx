import { theme, ThemeProvider } from '@celebrate-app/themes';
import { Story } from '@storybook/react';
import {
  cleanup,
  render,
  RenderOptions,
  waitFor,
} from '@testing-library/react-native';
import React, { ReactNode } from 'react';

const Providers = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export const customRender = (
  ui: React.ReactElement<any>,
  options?: RenderOptions
) => {
  return render(ui, { wrapper: Providers, ...options });
};

beforeEach(cleanup);
afterEach(cleanup);

export function testStory(displayName: string, TestComponent: Story) {
  test(`${displayName} snapshot test`, async () => {
    const comp = customRender(<TestComponent {...TestComponent.args} />);
    jest.runOnlyPendingTimers();

    await waitFor(() => {
      expect(comp).toBeTruthy();
      expect(comp.toJSON()).toMatchSnapshot();
    });
  });
}

export function testComponentSnapshot(
  TestComponent: React.ReactElement<
    any,
    string | React.JSXElementConstructor<any>
  >
) {
  test('Snapshot test', () => {
    const comp = customRender(TestComponent);
    expect(comp).toBeTruthy();
    expect(comp.toJSON()).toMatchSnapshot();
  });
}

export function testAllStories(stories: any) {
  Object.keys(stories).forEach((story) => {
    if (story !== 'default') {
      testStory(story, stories[story]);
    }
  });
}
