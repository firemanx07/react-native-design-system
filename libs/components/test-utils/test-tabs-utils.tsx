import { waitFor } from '@testing-library/react-native';

import { customRender } from './test-utils';

type renderTabsProps = {
  elem: JSX.Element;
  testIDs: string[];
};
export const renderTabs = ({ elem, testIDs }: renderTabsProps) => {
  const comp = customRender(elem);
  const tabs: ReturnType<typeof comp.getByTestId>[] = [];
  testIDs.forEach(testID => tabs.push(comp.getByTestId(testID)));
  return { comp, tabs };
};
export const testTabs = async (args: ReturnType<typeof renderTabs>) => {
  await waitFor(() => {
    expect(args.comp).toBeTruthy();
    args.tabs.forEach(tab => expect(tab).toBeDefined());
  });
};
