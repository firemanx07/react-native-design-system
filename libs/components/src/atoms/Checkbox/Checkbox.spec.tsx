import { testAllStories } from '../../../test-utils';
import * as stories from './Checkbox.stories';

describe('Checkbox tests', () => {
  testAllStories(stories);
});
