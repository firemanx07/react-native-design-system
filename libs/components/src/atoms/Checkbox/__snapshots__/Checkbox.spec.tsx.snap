// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox tests Checkbox snapshot test 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "minHeight": 600,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "borderColor": "#2C2C2C",
          "borderRadius": 4,
          "borderStyle": "dashed",
          "borderWidth": 1,
          "marginBottom": 24,
          "marginLeft": 24,
          "marginRight": 24,
          "marginTop": 24,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": 370,
        },
      ]
    }
    width={370}
  >
    <Text
      fontFamily="Rubik-Bold"
      fontSize={16}
      lineHeight={20}
      style={
        [
          {
            "color": "#2C2C2C",
            "fontFamily": "Rubik-Bold",
            "fontSize": 16,
            "lineHeight": 20,
          },
          {
            "backgroundColor": "#DEDFE0",
            "left": 16,
            "paddingHorizontal": 8,
            "position": "absolute",
            "top": -12,
          },
        ]
      }
    >
      CheckBox:
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "height": 20,
          "justifyContent": "center",
          "opacity": 1,
          "width": 20,
        }
      }
    >
      <test-file-stub
        fill="#49C4E9"
        height="100%"
        style={
          [
            {},
          ]
        }
        width="100%"
      />
    </View>
  </View>
  <View
    style={
      [
        {
          "borderColor": "#2C2C2C",
          "borderRadius": 4,
          "borderStyle": "dashed",
          "borderWidth": 1,
          "marginBottom": 24,
          "marginLeft": 24,
          "marginRight": 24,
          "marginTop": 24,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": 370,
        },
      ]
    }
    width={370}
  >
    <Text
      fontFamily="Rubik-Bold"
      fontSize={16}
      lineHeight={20}
      style={
        [
          {
            "color": "#2C2C2C",
            "fontFamily": "Rubik-Bold",
            "fontSize": 16,
            "lineHeight": 20,
          },
          {
            "backgroundColor": "#DEDFE0",
            "left": 16,
            "paddingHorizontal": 8,
            "position": "absolute",
            "top": -12,
          },
        ]
      }
    >
      Disabled:
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "height": 20,
          "justifyContent": "center",
          "opacity": 1,
          "width": 20,
        }
      }
    >
      <test-file-stub
        fill="#C7C7D2"
        height="100%"
        style={
          [
            {},
          ]
        }
        width="100%"
      />
    </View>
  </View>
  <View
    style={
      [
        {
          "borderColor": "#2C2C2C",
          "borderRadius": 4,
          "borderStyle": "dashed",
          "borderWidth": 1,
          "marginBottom": 24,
          "marginLeft": 24,
          "marginRight": 24,
          "marginTop": 24,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": 370,
        },
      ]
    }
    width={370}
  >
    <Text
      fontFamily="Rubik-Bold"
      fontSize={16}
      lineHeight={20}
      style={
        [
          {
            "color": "#2C2C2C",
            "fontFamily": "Rubik-Bold",
            "fontSize": 16,
            "lineHeight": 20,
          },
          {
            "backgroundColor": "#DEDFE0",
            "left": 16,
            "paddingHorizontal": 8,
            "position": "absolute",
            "top": -12,
          },
        ]
      }
    >
      inactive:
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "height": 20,
          "justifyContent": "center",
          "opacity": 1,
          "width": 20,
        }
      }
    >
      <View
        size={20}
        style={
          [
            {
              "borderColor": "#C7C7D2",
              "borderRadius": 6.2,
              "borderWidth": 2,
              "height": "100%",
              "width": "100%",
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
